src/components/RequestExecutorV2.bs

```diff
@@ -0,0 +1,232 @@
+  transferObject.asyncPostFromString(body)
+end function
+
+sub processResponse(transferObjectResponse)
+  if transferObjectResponse = invalid or transferObjectResponse.getSourceIdentity() = invalid then
+    throw "Can not process a transfer object response without a transfer object identity"
+  end if
+  transferObjectId = stri(transferObjectResponse.getSourceIdentity()).trim()
+  finishedTransferObject = getTransferObjectForId(transferObjectId)
+  if finishedTransferObject = invalid then
+    throw `Can not process a transfer object response with an identity of ${transferObjectId} because no transfer object with that identity exists. Something is wrong with the transfer object pool.`
+  end if
+  m.availableTransferObjects.push(finishedTransferObject)
+  if DEBUG then ?`Receiving response
+  availableTransferObjects: ${m.availableTransferObjects.count()}
+  requests in queue: ${m.requestQueue.count()}
+  requests in flight: ${m.requestsInFlight.count()}
+  `
+  response = transferObjectResponse.getString()
+  if DEBUG_NOISY then ?"ResponseCode "transferObjectResponse.getResponseCode()
+  if Debug_NOISY then ?"Response "response
+  responseObject = parseJson(response)
+  originalRequest = m.requestsInFlight[transferObjectId]
+  m.requestsInFlight.delete(transferObjectId)
+  if originalRequest.callback = invalid then return ' No callback to invoke
+  invokeCallback(originalRequest.callback, responseObject)
+end sub
+
+function getTransferObjectForId(transferObjectId as STRING)
+  for i = 0 to m.transferPool.count() - 1
+    mTransferObject = m.transferPool[i]
+    if stri(mTransferObject.getIdentity()).trim() = transferObjectId then return mTransferObject
+  end for
+  return invalid
+end function
+
+sub invokeCallback(callback as OBJECT, responseObject as OBJECT)
+  if callback = invalid then return
+  if callback.componentName = invalid or callback.functionName = invalid
+    throw `Callback provided but missing either componentName or functionName.
+    componentName: ${callback.componentName}
+    functionName: ${callback.functionName}`
+    return
+  end if
+  component = findComponent(callback.componentName)
+  callbackName = callback.functionName
+  if DEBUG_NOISY then ?`invokeCallback: ${component.subtype()}#${callback}`
+  callbackResult = component.callFunc(callbackName, responseObject)
+  if callbackResult = invalid then
+    throw `Callback ${callbackName} on ${component.subtype()} returned invalid. Callbacks must return something (e.g. true) to indicate a function signature match.`
+  end if
+end sub
+
+' Only checks for a component one level deep from m.top.getScene() since that is our
+' internal use case. If you need to find components nested more deeply than that you
+' could search recursively. Please PR if you do!
+function findComponent(componentName as STRING)
+  container = m.top.getScene()
+  scene = invalid
+  for i = 0 to container.getChildCount() - 1
+    if container.getChild(i).subType() = componentName
```

Comment:
This entire flow feels very strange, and not at all safe. You're just looking for the first component with this type. What if it was `rectangle` and there are 100 of them? I can't imagine this approach being overly useful outside of your specific use case.

I think it would be a much better approach if you have the caller pass in the component reference itself. 

Or, perhaps this task could actually create the component and pass it back. So you would have something like:
```
callback = {
    componentName: "YourCustomComponent",
    functionName: "doSomething"
}

component = createObject("roSGNode", callback.componentName)
component.callfunc(callback.functionName)

'then return the instance of that component

Reply:
Ah, fair point! Our group components are long-lived and generally the only instance attached to the scene. 

Creating the component wouldn't work for us, but passing a reference makes sense and is even more intuitive. 

How would we get a ref to the component from w/i it? AFAIK there's no `this`. Maybe pass a ref of itself into the component from where it's created?

Reply:
Another alternative wooumd be setting an ID on the component's interface and doing a "findbyid"

Reply:
yeah, that would be slightly better. However, IDs are not enforced to be unique across the app. But i guess that would be up to the developer to enforce uniqueness.

Reply:
Could use [GUID](https://developer.roku.com/docs/references/brightscript/interfaces/ifdeviceinfo.md#getrandomuuid-as-string)s. Did you have thoughts about an easy way to pass a reference to the component? This seems like the cleanest approach.